### PR TITLE
Update path where API spec is served to allow unauthenticated requests

### PR DESCRIPTION
**Pull request by ecp88**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -57,7 +57,7 @@ async fn playground_ui() -> impl actix_web::Responder {
                   <body>
 
                     <elements-api
-                      apiDescriptionUrl="{base_path}/spec/v3.json"
+                      apiDescriptionUrl="/eapi/spec/v1/v3.json"
```
Comment (roshaans) [inline src/main.rs]: We should get version name from from env var so that in the future, we only need to change one env var to get everything working.

Comment (roshaans) [reply src/main.rs]:
```
fn get_api_version() -> String {
    std::env::var("API_VERSION").unwrap_or_else(|_| "v1".to_string())
}
```
```
let api_version = get_api_version();
```

```suggestion
                      apiDescriptionUrl="/eapi/spec/{api_version}/v3.json"
```

File: src/main.rs [hunk 2]
```diff
@@ -187,8 +187,8 @@ We would love to hear from you on the data APIs you need, please leave feedback
         app = app.configure(modules::ft::register_services);
         app = app.configure(modules::nft::register_services);
 
-        app.with_json_spec_at(format!("{base_path}/spec/v2.json").as_str())
```
Comment (roshaans) [inline src/main.rs:190]: Cant add change to the lines above, but on line 156, we should use the same function as above `get_api_version` so we get the following:

```
 let api_version = get_api_version();
 spec.base_path = Some(format!("eapi/{api_version}"));
```

File: src/main.rs [hunk 3]
```diff
@@ -187,8 +187,8 @@ We would love to hear from you on the data APIs you need, please leave feedback
...
-            .with_json_spec_v3_at(format!("{base_path}/spec/v3.json").as_str())
+        app.with_json_spec_at(format!("/eapi/spec/v1/v2.json").as_str())
```
Comment (roshaans) [inline src/main.rs]:
```suggestion
        app.with_json_spec_at(format!("/eapi/spec/{api_version}/v2.json").as_str())
```

**Review by telezhnaya — CHANGES_REQUESTED** [review]
Why do we need these changes? How should it help to allow unauthenticated requests?
Why can't we just set env variable `BASE_PATH` to `eapi/spec/v1`?
What if we want to change this prefix?

File: src/main.rs (hunk 2 above)
Comment (ecp88) [reply src/main.rs:190]: what if `eapi` in the path changes? If we are going to have another env var anyways then lets get a `spec_base_path` that we use only to know where to serve the spec. 
PTAL

File: src/main.rs (hunk 1 above)
Comment (ecp88) [reply src/main.rs]: Added a new env var `spec_base_path `

**Comment by ecp88** [comment]
> Why do we need these changes? How should it help to allow unauthenticated requests? Why can't we just set env variable `BASE_PATH` to `eapi/spec/v1`? What if we want to change this prefix?

- We need to serve the spec in a different path. The `base_path` prefix requires authentication on all paths under it. The `spec_path` will not. So, the home page redirects and loads a `spec_path` that not requires authentication and is able to serve the playground.

**Review by telezhnaya — APPROVED** [review]
I can't test it, but the changes look fine to me

**Pull request merged**